### PR TITLE
adding vmss extension demo

### PR DESCRIPTION
**Pull request by zikalino**
## Purpose
e2e demo for vmss extension

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## How to Test
*  Install ansible

```bash
$ pip install ansible[azure]
```

*  [Optional] Install ansible role

```bash
$ ansible-galaxy install Azure.azure_preview_modules
$ pip install -r ~/.ansible/roles/Azure.azure_preview_modules/files/requirements-azure.txt
```

*  Get the sample playbook

```
git clone https://github.com/Azure-Samples/ansible-playbooks.git
cd ansible-playbooks
git checkout [branch-name]
```

* Test the code

```bash
ansible-playbook <filename>.yml
```

## What to Check
Verify that the playbook is successfully run.

### CODE REVIEW

File: vmss_extension/01-create-vmss.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Description
+# ===========
+# This playbook creates a Virtual Machine Scale Set with of the shelf Ubuntu 16.04 image.
```
Comment (yungezz) [inline vmss_extension/01-create-vmss.yml]: shelf -> image?

File: vmss_extension/03-update-instances.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Description
+# ===========
+# This playbook updates Virtual Machine Scaleset instances.
```
Comment (yungezz) [inline vmss_extension/03-update-instances.yml]: maybe clear information on why need upgrades the vmss instances. it's "upgrade" instead of "update""

Comment (yungezz) [reply vmss_extension/03-update-instances.yml]: so why need upgrade here? it's requrired after install custom script extension?

**Comment by yungezz** [comment]
general good, some minor questions.

File: vmss_extension/03-update-instances.yml (hunk 2 above)
Comment (zikalino) [reply vmss_extension/03-update-instances.yml]: no, existing instances need "upgrade" only new instances will be automatically flashed with new image + extensions. mechanism is exactly the same as changing image.

**Pull request merged**